### PR TITLE
Story/hfi calc freeze first col/1220

### PR DESCRIPTION
**Pull request by andrea-williams**
# Test Links:
[Percentile Calculator](https://wps-pr-1580.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1580.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1580.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1580.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1580.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=1108&f=c5&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN#state=2ec784ca-c46a-49d0-b2b3-1cf32a9015a2&session_state=7d9447c8-db66-4661-b4cb-03d2ac0d1d8f&code=32292df4-2bdf-4f90-a4a8-c8dbcda682a9.7d9447c8-db66-4661-b4cb-03d2ac0d1d8f.2b63f390-f3dc-43ae-89f2-016453863476)
[Fire Behaviour Advisory](https://wps-pr-1580.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1580.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: web/src/app/theme.ts [hunk 1]
```diff
@@ -99,6 +105,15 @@ export const fireTableStyles = createStyles({
   },
   spaceHeader: {
     border: 'none'
+  },
+  unstickyHeaderCell: {
```
Comment (conbrad) [inline web/src/app/theme.ts:109]: Minor: is unsticky a word? I think it's nonsticky.

**Review by Sybrand — APPROVED** [review]
It works! (I am getting more and more worried that we're building a very fragile UI though)

File: web/src/features/hfiCalculator/components/FireCentreCells.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import React from 'react'
+
+interface FireCentreCellsProps {
+  centre: FireCentre
+  testId?: string
+}
+
+const useStyles = makeStyles({ ...fireTableStyles })
+
+const FireCentreCells = (props: FireCentreCellsProps) => {
```
Comment (conbrad) [inline web/src/features/hfiCalculator/components/FireCentreCells.tsx:14]: Minor: Singular cell, `FireCentreCells` -> `FireCentreCell`

File: web/src/features/hfiCalculator/components/WeeklyViewTable.tsx [hunk 3]
```diff
@@ -138,9 +150,30 @@ export const WeeklyViewTable = (props: Props): JSX.Element => {
...
+                            {area.name}
+                          </TableCell>
+                        </StickyCell>
+                        <TableCell
+                          className={`${classes.planningArea} ${classes.unstickyHeaderCell}`}
+                        ></TableCell>
+                        <StickyCell
+                          left={230}
+                          zIndexOffset={10}
+                          backgroundColor={'#e9ecf5'}
```
Comment (conbrad) [inline web/src/features/hfiCalculator/components/WeeklyViewTable.tsx:169]: Styling in 2 places here, `backgroundColor` and `className`. Can we add the background color to a new class so we save ourselves maintenance pain in the future?

File: web/src/features/hfiCalculator/components/WeeklyViewTable.tsx [hunk 4]
```diff
@@ -138,9 +150,30 @@ export const WeeklyViewTable = (props: Props): JSX.Element => {
                         key={`zone-${areaName}`}
                         data-testid={`zone-${areaName}`}
                       >
-                        <TableCell className={classes.planningArea} colSpan={5}>
-                          {area.name}
-                        </TableCell>
+                        <StickyCell
+                          left={0}
+                          zIndexOffset={10}
+                          backgroundColor={'#e9ecf5'}
```
Comment (conbrad) [inline web/src/features/hfiCalculator/components/WeeklyViewTable.tsx:156]: For other parameters of sticky cells like `left`, etc, they are usually non-uniform. This `backgroundColor` is repeated a bunch though and probably can be named as a variable.

**Review by conbrad — CHANGES_REQUESTED** [review]
Looks good, just a few minor comments.

Also, there is a small doubling of borders in one specific case, see screenshot below:
Screen Shot 2021-12-01 at 9 34 38 AM

**Review by conbrad — APPROVED** [review]

**Comment by tessmgood** [comment]
> It works! (I am getting more and more worried that we're building a very fragile UI though)

Let's declare fragile UI our next Human Experience agenda. I might even extend it to 60 minutes - special edition.

**Review by tessmgood — APPROVED** [review]
nice. also noticing the border weirdness Conor pointed out, but very demo-able. Thanks for the UI enhancements!

**Comment by andrea-williams** [comment]
I've made the changes suggested by @conbrad regarding variable naming.

Also played around slightly with the borders in the tables, but I haven't been able to reproduce the double border issue that Conor screenshotted. There is a double-border in the weekly table between the grass cure col & the ROS for the first day of the prep cycle - I think it's too much of a risk to try to fix that now so close to sprint review. We know that our border/separators in our table need more attention anyway (based on what @tessmgood has told me), so I propose we leave this particular detail until that future ticket that's dedicated to strategically adding borders within tables.

**Pull request merged**